src/inspect_ai/scorer/_metric.py

```diff
@@ -53,6 +55,46 @@
 the `Value` as a simple scalar of various types.
 """
 
+UNCHANGED: Literal["UNCHANGED"] = "UNCHANGED"
+"""Sentinel value to indicate an unchanged field in score edits."""
+
+
+class ProvenanceData(BaseModel):
+    """Metadata about who made an edit and why."""
+
+    timestamp: datetime = Field(default_factory=datetime.now)
+    """Timestamp when the edit was made."""
+
+    author: str
+    """Author who made the edit."""
+
+    reason: str | None = Field(default=None)
+    """Reason for the edit."""
+
+    metadata: dict[str, Any] = Field(default_factory=dict)
+    """Additional metadata about the edit."""
+
+
+class ScoreEdit(BaseModel):
+    """A single edit to a score."""
+
+    model_config = ConfigDict(validate_assignment=True)
```

Comment:
Ok last question - is this necessary? This appears to be the only place in the system we're doing this validation on assignment, so think we should omit unless there is a good reason for adding it here specifically....